### PR TITLE
Pareto chart implementation

### PR DESCRIPTION
**Pull request by smoothdeveloper**
@kMutagene, this is for #423, please let us know what changes you want.

Here is snippet of rendered documentation:

image

cc: @rockfaith75

### CODE REVIEW

**Comment by smoothdeveloper** [comment]
@kMutagene if you have an idea how to adjust so that the legend at the top right doesn't overlay on top of the "100%" label, please let me know.

I'm wondering also if we should have optional parameter that would take the separate plots to alter them:

```fsharp
type ParetoChartPart =
| ColumnPart
| ParetoLinePart

Chart.Pareto(...
  , customizeChartPart=[
     ColumnPart, (fun chart -> chart |> Chart.with...)
    ]
 )
```

This technique could be used in the different composed charts.

File: src/Plotly.NET/ChartAPI/Chart2D.fs [hunk 1]
```diff
@@ -1547,7 +1547,67 @@ module Chart2D =
...
+                Seq.scan (+) 0. orderedValues
+                |> Seq.skip 1
+               
+            let paretoValues = 
+               Seq.zip orderedLabels cumulativeSum
+               |> Seq.map (fun (label,value) -> label, value / sum * 100.) 
+               
+            let bars = Chart.Column(Seq.zip orderedLabels orderedValues,?Name=Name)
+              
+            let points =
```
Comment (kMutagene) [inline src/Plotly.NET/ChartAPI/Chart2D.fs]: there is no need for 2 traces for points and lines here, you can show markers on a line chart with `ShowMarkers = true`.

File: src/Plotly.NET/ChartAPI/Chart2D.fs [hunk 2]
```diff
@@ -1547,7 +1547,67 @@ module Chart2D =
...
+                    , ShowLegend = true
+                ) 
+                |> Chart.withAxisAnchor (Y = 2)
+                   
+            [bars;lines;points] 
+            |> Chart.combine 
+            |> Chart.withYAxisStyle (
+                    ?TitleText = Label
+                    , Id       = StyleParam.SubPlotId.YAxis 1
+                    , ShowGrid = false
```
Comment (kMutagene) [inline src/Plotly.NET/ChartAPI/Chart2D.fs:1596]: I usually expose settings like this as method argument, and setting sensible default values.

**Review by kMutagene — APPROVED** [review]
Hi guys, this looks great. This is in general ready to merge. if you want to improve a bit you can use a single line chart instead of one point and one line chart, and expose more customizability via optional arguments. Otherwise, LGTM!

**Comment by kMutagene** [comment]
Also, a version using labels and values as separate sequences instead of tupled values would be nice as well, but not necessary. This would have the advantage that we do not need a custom tuple type for the C# version, and instead expose the method using separate sequences (this is what i did with all other C# methods)

**Comment by smoothdeveloper** [comment]
@kMutagene thanks for the feedback, we'll go over the suggestions / adjustments probably over Sunday.

**Comment by smoothdeveloper** [comment]
@kMutagene I made adjustments, the main remarks I have:

* overloads with separate `seq` doesn't preclude user to pass mismatched data
* removing the points chart makes the markers render in the same color as the line itself, the good aspect is that it removes needless `trace2` information on hover

Let see if people need specific customizations.

**Comment by kMutagene** [comment]
> overloads with separate seq doesn't preclude user to pass mismatched data

Can you give an example of mismatched data? I think i do not understand exactly.

> removing the points chart makes the markers render in the same color as the line itself

Can also be prevented by styling the markers of the line chart accordingly, but i think this can be done after merging this PR.

In general, this is a really high quality PR as it even includes documentation, thanks @smoothdeveloper and @rockfaith75 !

The only thing needed for release is addition of some tests, but since it is quite complicated to set up tests in this repo I will take care of that. Thanks again!

**Comment by smoothdeveloper** [comment]
> Can you give an example of mismatched data? I think i do not understand exactly.

like

```fsharp
let values = [1.;2.;3.]
let labels = ["a";"b"] // missing one label
Chart2D.Pareto(labels, values)
```

Thanks for the review and we'll be on the lookout for how the tests are added for future contributions.

**Comment by kMutagene** [comment]
>like
let values = [1.;2.;3.]
let labels = ["a";"b"] // missing one label
Chart2D.Pareto(labels, values)

I see, but this is handled by plotly.js by simply only displaying data that has both dimensions, so we overall do not really care about this across the lib.

**Pull request merged**